packages/financial-templates-lib/src/proxy-transaction-handler/DSProxyManager.js

```diff
@@ -0,0 +1,150 @@
+    this.isAddress = this.web3.utils.isAddress;
+
+    // Multiplier applied to Truffle's estimated gas limit for a transaction to send.
+    this.GAS_LIMIT_BUFFER = 1.25;
+  }
+
+  getDSProxyFactoryAddress() {
+    return this.dsProxyFactoryAddress;
+  }
+
+  getDSProxyAddress() {
+    return this.dsProxyAddress;
+  }
+
+  // Load in an existing DSProxy for the account EOA if one already exists OR create a new one for the user.
+  async initializeDSProxy() {
+    this.logger.debug({
+      at: "DSProxyManager",
+      message: "Initalizing...Looking for existing DSProxies or deploying a new one for the provided EOA",
+      dsProxyFactoryAddress: this.dsProxyFactoryAddress
+    });
+
+    if (this.dsProxy && this.dsProxyAddress) this.dsProxyAddress;
+    const fromBlock = await getFromBlock(this.web3);
+    const events = await this.dsProxyFactory.getPastEvents("Created", {
+      fromBlock,
+      filter: { owner: this.account }
+    });
+
+    // The user already has a DSProxy deployed. Load it in from the events.
+    if (events.length > 0) {
+      this.dsProxyAddress = events[events.length - 1].returnValues.proxy;
+      this.dsProxy = new web3.eth.Contract(this.dsProxyAbi, this.dsProxyAddress);
+      this.logger.debug({
+        at: "DSProxyManager",
+        message: "DSProxy has been loaded in for the EOA",
+        dsProxyAddress: this.dsProxyAddress,
+        tx: events[events.length - 1].transactionHash,
+        account: this.account
+      });
+    }
+
+    // The user does not yet have a DSProxy. Create them one.
+    if (events.length == 0) {
+      this.logger.debug({
+        at: "DSProxyManager",
+        message: "No DSProxy found for EOA. Deploying new DSProxy",
+        account: this.account
+      });
+      const dsProxyCreateTx = await this.dsProxyFactory.methods.build().send({
+        from: this.account,
+        gasPrice: this.gasEstimator.getCurrentFastPrice()
+      });
+      this.dsProxyAddress = dsProxyCreateTx.events.Created.returnValues.proxy;
+      this.dsProxy = new web3.eth.Contract(this.dsProxyAbi, this.dsProxyAddress);
+      this.logger.info({
+        at: "DSProxyManager",
+        message: "DSProxy has been deployed for the EOA 🚀",
+        dsProxyAddress: this.dsProxyAddress,
+        tx: dsProxyCreateTx.transactionHash,
+        account: this.account
+      });
+    }
+    return this.dsProxyAddress;
+  }
+  // Encode with `yourTruffleContractInstance.yourMethod(params1,param2).encodeABI()
+  async callFunctionOnExistingLibrary(libraryAddress, callData) {
+    assert(this.isAddress(libraryAddress), "Library address must be valid address");
+    assert(typeof callData === "string", "Call data must be a string");
+    this.logger.debug({
+      at: "DSProxyManager",
+      message: "Executing function on deployed library",
+      libraryAddress,
+      callData
+    });
+
+    const executeTransaction = await this.dsProxy.methods["execute(address,bytes)"](libraryAddress, callData).send({
```

Comment:
i feel like since this is relying on initializing the class, this should be calling a getter which throws if `this.dsProxy` does not exist.  same with other references to `this.dsProxy`. thoughts?

Reply:
I added an assert to check that the DSProxy is set.